crates/sui-adapter/src/execution_engine.rs

```diff
@@ -224,6 +224,37 @@ fn execute_transaction<
                 ))
             }
         };
+        if execution_result.is_ok() {
+            let written_objects_size = temporary_store.written_objects_size();
+
+            match check_limit_by_meter!(
+                !gas_status.is_unmetered(),
+                written_objects_size,
+                protocol_config.max_size_written_objects(),
```

Comment:
I think this code will crash if run on a validator that's at protocol version 1?

Reply:
I think we need to feature flag this or something? Is there a way to check that it is `Some(_)`, could be nice since the config will serve both as the config and the flag. 
@mystenmark, have any suggestions?

Reply:
> I think this code will crash if run on a validator that's at protocol version 1?

I thought that was by design right? @mystenmark

Reply:
It's "by design" that it crashes, in the sense that there is definitely a bug if you try to access this value in an old protocol version, so in that case you'd want to crash instead of fork. However it's not "by design" in the sense that you probably don't want your PR to intentionally make validators crash if they are running a new binary on a network with older protocol versions.

Reply:
So whats the best way to phase this in?

Reply:
Yes, when you are adding a new feature like this, you need to add the new constants, and add a feature flag. If you access the constants during a protocol version in which they aren't defined, we want to crash, because that indicates that you didn't properly gate the new code.

If we didn't crash, then the validator that is reading the new constants will almost certainly fork from any validator that doesn't have the new binary.